### ISSUE TITLE
Key not found: lastModifiedDateTime

### ISSUE THREAD

**Issue by GeegC** [skilion/onedrive#351]
My onedrive (version onedrive v1.0.1-29-gc7e0930), which I am using on a business account, won't sync my files properly and says it can't find the key lastModifiedDateTime for the files I have changed since the last sync.

When running onedrive --verbose I get the following:

Loading config ...
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /home/georgia/OneDrive
Initializing the Synchronization Engine ...
Applying changes of 01UCFEKYN6Y2GOVW7725BZO354PWSELRRZ
Processing 01UCFEKYN6Y2GOVW7725BZO354PWSELRRZ root
Root
Processing 01UCFEKYJKZR7TOO3QEVG2PXSHFCTCGDLR A-PhD
Folder
The item has not changed
Processing 01UCFEKYJ6VN7LUNLGYNFKD37LDA6A2LVA Bioinformatics
Folder
The item has not changed
Key not found: lastModifiedDateTime
Applying changes of 01UCFEKYN6Y2GOVW7725BZO354PWSELRRZ
Processing 01UCFEKYN6Y2GOVW7725BZO354PWSELRRZ root
Root
Processing 01UCFEKYJKZR7TOO3QEVG2PXSHFCTCGDLR A-PhD
Folder
The item has not changed
Processing 01UCFEKYJ6VN7LUNLGYNFKD37LDA6A2LVA Bioinformatics
Folder
The item has not changed
Key not found: lastModifiedDateTime
Applying changes of 01UCFEKYN6Y2GOVW7725BZO354PWSELRRZ
Processing 01UCFEKYN6Y2GOVW7725BZO354PWSELRRZ root
Root
Processing 01UCFEKYJKZR7TOO3QEVG2PXSHFCTCGDLR A-PhD
Folder
The item has not changed
Processing 01UCFEKYJ6VN7LUNLGYNFKD37LDA6A2LVA Bioinformatics
Folder
The item has not changed
std.json.JSONException@std/json.d(494): Key not found: lastModifiedDateTime
----------------
??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], ulong, const(char[])) [0x5e370f]
??:? pure @safe inout(std.json.JSONValue)* std.exception.enforce!(std.json.JSONException, inout(std.json.JSONValue)*).enforce(inout(std.json.JSONValue)*, lazy const(char)[], immutable(char)[], ulong) [0x5e377a]
??:? inout pure ref @safe inout(std.json.JSONValue) std.json.JSONValue.opIndex(immutable(char)[]) [0x59eadd]
src/sync.d:47 itemdb.Item sync.makeItem(ref const(std.json.JSONValue)) [0x578325]
src/sync.d:196 void sync.SyncEngine.applyDifference(std.json.JSONValue, immutable(char)[], bool) [0x579202]
src/sync.d:177 void sync.SyncEngine.applyDifferences(immutable(char)[], const(char)[]) [0x579047]
src/sync.d:146 void sync.SyncEngine.applyDifferences() [0x578d25]
src/main.d:190 void main.performSync(sync.SyncEngine) [0x5691c7]
src/main.d:118 _Dmain [0x568d24]

I have rebooted the computer and refreshed the online onedrive site before running this command.

Thanks :)

**Comment by abraunegg** [comment]
@GeegC @setras 

The original author appears to have abandoned this code. To resolve your issues, you need to be using #314 to get the fixes. There are 2 ways to get the latest code:

Option 1:
```
git clone https://github.com/skilion/onedrive.git
git fetch origin pull/314/head:pr314
git checkout pr314
make
make install
```

Option 2:

```
git clone https://github.com/abraunegg/onedrive.git
make
make install
```

**Comment by GeegC** [comment]
Hi abraunegg,

Thanks very much.  I have merged the latest version of your fork into my repo.

Unfortunately I still see an error:

`Key not found: lastModifiedDateTime
Key not found: lastModifiedDateTime
std.json.JSONException@std/json.d(494): Key not found: lastModifiedDateTime
----------------
??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], ulong, const(char[])) [0x5e370f]
??:? pure @safe inout(std.json.JSONValue)* std.exception.enforce!(std.json.JSONException, inout(std.json.JSONValue)*).enforce(inout(std.json.JSONValue)*, lazy const(char)[], immutable(char)[], ulong) [0x5e377a]
??:? inout pure ref @safe inout(std.json.JSONValue) std.json.JSONValue.opIndex(immutable(char)[]) [0x59eadd]
src/sync.d:47 itemdb.Item sync.makeItem(ref const(std.json.JSONValue)) [0x578325]
src/sync.d:196 void sync.SyncEngine.applyDifference(std.json.JSONValue, immutable(char)[], bool) [0x579202]
src/sync.d:177 void sync.SyncEngine.applyDifferences(immutable(char)[], const(char)[]) [0x579047]
src/sync.d:146 void sync.SyncEngine.applyDifferences() [0x578d25]
src/main.d:190 void main.performSync(sync.SyncEngine) [0x5691c7]
src/main.d:118 _Dmain [0x568d24]
`

Thanks again

**Comment by abraunegg** [comment]
@GeegC 

Either your merge or something else is wrong. As your clone / repository is not available I cannot check - but this is certainly solved in #314

**Comment by abraunegg** [comment]
@GeegC 
Any update?

**Comment by abraunegg** [comment]
@GeegC 
Any update?

**Comment by abraunegg** [comment]
@GeegC 
Any update?

**Comment by abraunegg** [comment]
@GeegC
Any update?

**Comment by GeegC** [comment]
I've been on holiday but will try and find time to have a look at this in the next few days. Thanks :)

**Comment by saschpe** [comment]
same here

**Comment by abraunegg** [comment]
@saschpe 
The original author appears to have abandoned this code. To resolve your issues, you need to be using #314 to get the fixes or use my fork. There are 2 ways to get the latest code:

Option 1:
```
git clone https://github.com/skilion/onedrive.git
git fetch origin pull/314/head:pr314
git checkout pr314
cd onedrive
make
make install
```

Option 2:

```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
make
make install
```

**Comment by abraunegg** [comment]
Considering this issue closed due to no response from original submitter.

If you have further issues please or come back to this, open a new case here: https://github.com/abraunegg/onedrive

**Comment by skilion** [comment]
Fixed in 3d6262b3ee7f371d6ee99ea16037f41b8b00dc71

**Comment by GeegC** [comment]
Hi @skilion,

Thanks for this fix.  Unfortunately I now have a new error.

As soon as I type "onedrive" in order to sync I get the following:

```
JSONValue is not an object
JSONValue is not an object
std.json.JSONException@std/json.d(260): JSONValue is not an object
----------------
??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], ulong, const(char[])) [0x5e370f]
??:? pure @safe bool std.exception.enforce!(std.json.JSONException, bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x5e3689]
??:? inout pure @property @trusted inout(std.json.JSONValue[immutable(char)[]]) std.json.JSONValue.objectNoRef() [0x59e8c9]
/usr/include/dmd/phobos/std/json.d:599 const pure @safe const(std.json.JSONValue)* std.json.JSONValue.opBinaryRight!("in").opBinaryRight(immutable(char)[]) [0x57314f]
src/sync.d:184 void sync.SyncEngine.applyDifferences(immutable(char)[], const(char)[]) [0x579169]
src/sync.d:146 void sync.SyncEngine.applyDifferences() [0x578d25]
src/main.d:190 void main.performSync(sync.SyncEngine) [0x5691c7]
src/main.d:118 _Dmain [0x568d24]
```
Thank you!
